### PR TITLE
update(libsinsp): add k8s node name validation

### PR DESCRIPTION
**Pull request by loresuso**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area driver-kmod

> /area driver-ebpf

> /area libscap

/area libsinsp

> /area tests

> /area proposals

**What this PR does / why we need it**:

This PR tries to introduce a very simple validation check on the name of the k8s node that is used as a filter when retrieving pods metadata. As discussed [here](https://github.com/falcosecurity/libs/issues/136), autodetection of the node name may fail in some edge cases. For this reason, I followed the suggestion to just check whether the node name actually corresponds to the name of a node in the cluster. I am aware that this is not the perfect solution, but it at least helps any consumer of the `libs` to understand if Kubernetes metadata are correctly retrieved and used, instead of silently failing, in many circumstances.

**Which issue(s) this PR fixes**:

Fixes https://github.com/falcosecurity/libs/issues/136

**Special notes for your reviewer**:

Since the proposed solution is intrinsically not optimal, I opted for a lazy validation approach. The idea is to let everything initialize with the node name passed. Right after, the k8s state will be filled with the vector of nodes in the cluster and that information can be used to perform the check. This also allows avoiding a new call to the API server to list the names of the nodes in the cluster, by reusing the one we already have.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libsinsp/sinsp.cpp
Comment (FedeDP) [inline userspace/libsinsp/sinsp.cpp:2273]: Is m_k8s_node_name_validation_state used elsewhere? I cannot find anything else about it!

Comment (loresuso) [reply userspace/libsinsp/sinsp.cpp:2273]: That is meant to be used by libs consumers! Check also [this one](https://github.com/falcosecurity/falco/pull/1823) for an example. Please let me know what do you think about this!

Comment (FedeDP) [reply userspace/libsinsp/sinsp.cpp:2273]: Uh i see! I though this was self contained inside libs :) 
Thank you!

**Review by FedeDP — APPROVED** [review]

**Comment by loresuso** [comment]
Hi @leogr,
yep, I think you are right, mainly because I don't really much like where I put `m_k8s_node_name_validation_state` check in Falco. I'm modifying it by throwing an exception in sinsp, will ping you when asap when it's done!

**Comment by loresuso** [comment]
Should be ok now! Let me know when you have time, @FedeDP @leogr  😃

**Comment by FedeDP** [comment]
I like this! Simple and effective! :rocket:

**Review by leogr — COMMENTED** [review]
I definitively like this. I have just need to test it a big before giving my approval.

Thank you!

**Review by leogr — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]

**Pull request merged**
